### PR TITLE
[0.21] test: Backports

### PR DESCRIPTION
**Pull request by maflcko**
Backport tests

### CODE REVIEW

**Comment by fanquake** [comment]
Should/can this include a backport for #21489?

**Comment by MarcoFalke** [comment]
Good idea. Done

File: src/test/fuzz/versionbits.cpp [hunk 1]
```diff
@@ -343,3 +349,4 @@ void test_one_input(const std::vector<uint8_t>& buffer)
         }
     }
 }
+} // namespace
```
Comment (ajtowns) [inline src/test/fuzz/versionbits.cpp:352]: This `} // namespace` line needs to be removed for the backport.

Comment (MarcoFalke) [reply src/test/fuzz/versionbits.cpp]: Correct. Fixed.

**Comment by vasild** [comment]
ACK 4546ac5af1c4105a0cf78a407fefcb189822bcf5

I verified that the corresponding commits are same in `master` and in the backport (modulo mechanical diffs due to diverged code bases).

**Review by vasild — APPROVED** [review]

**Comment by MarcoFalke** [comment]
Moved milestone due to lack of interest

**Comment by jnewbery** [comment]
ACK 4546ac5af1c4105a0cf78a407fefcb189822bcf5

Verified cherrypick/rebase.

**Comment by MarcoFalke** [comment]
Rebased

**Comment by jnewbery** [comment]
ACK b8af67eeefc9fc9622f839ec8919b7391d91bf6f

Verified rebase.

**Review by vasild — APPROVED** [review]
ACK b8af67eeefc9fc9622f839ec8919b7391d91bf6f

Part of the last commit is already applied via b529222ad18f7facbaff394455875b4aa65d653e.

**Comment by ajtowns** [comment]
ACK b8af67eeefc9fc9622f839ec8919b7391d91bf6f ; visually compared individual commits to originals, checked original commits are in master

**Pull request merged**